src/components/graph-view.js

```diff
@@ -416,18 +416,21 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
         this.removeEdgeElement(edge);
       });
 
-      // remove node
+      // remove node - timeout helps for removing async rendered nodes
+      let removeNodeTimeout = null;
+      const nodeIdToRemove = `node-${nodeId}-container`;
+      const removeNode = () => {
+        clearTimeout(removeNodeTimeout);
+        GraphUtils.removeElementFromDom(this.entities, nodeIdToRemove);
+      };
+
+      removeNodeTimeout = setTimeout(removeNode, 200);
```

Comment:
Could we just do: 
`setTimeout(() => GraphUtils.removeElementFromDom(this.entities, nodeIdToRemove), 200)`
?